tests/conftest.py

```diff
@@ -32,26 +56,70 @@
 
 
 def _calc_mat_vec_wavefunction(
-    block: Primitive, n_qubits: int, init_state: torch.Tensor, values: dict = {}
+    block: Primitive | Sequence,
+    init_state: torch.Tensor,
+    values: dict = {},
+    full_support: tuple | None = None,
 ) -> torch.Tensor:
     """Get the result of applying the matrix representation of a block to an initial state.
 
     Args:
         block: The black operator to apply.
-        n_qubits: Number of qubits in the circuit.
         init_state: Initial state to apply block on.
         values: Values of parameter if block is parametric.
 
     Returns:
         Tensor: The new wavefunction after applying the block.
 
     """
-    mat = block.tensor(n_qubits=len(block.qubit_support), values=values)
+    mat = block.tensor(values=values, full_support=full_support)
+    qubits = block.qubit_support if full_support is None else full_support
     return apply_operator(
-        init_state, mat, qubits=block.qubit_support, n_qubits=n_qubits
+        init_state,
+        mat,
+        qubits=qubits,
     )
 
 
+def _get_op_support(op: type[Primitive] | type[Parametric], n_qubits: int) -> tuple:
```

Comment:
Maybe it would be useful to turn these functions into `@fixture` since they are used in tests ?

Reply:
Well I wrote them more as helper functions. Maybe there would be some way to turn them more into fixtures... For now I just made a new `helpers.py` file not to bloat `conftest.py` with non-fixtures.